3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2012-2015 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+*/
+package com.snowplowanalytics
+package snowplow
+package enrich
+package common
+package enrichments
+package registry
+
+import utils.MapTransformer._
+import java.lang.{Integer => JInteger}
+import java.math.{BigDecimal => JBigDecimal}
+import java.lang.{Byte => JByte}
+// Java
+import java.net.URI
+import java.net._
+
+// Maven Artifact
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// json4s
+import org.json4s.JValue
+
+// Iglu
+import iglu.client.SchemaKey
+import iglu.client.validation.ProcessingMessageMethods._
+
+// Joda-Money
+import org.joda.money.{Money}
+import org.joda.time.{DateTime, DateTimeZone}
+
+//Scala-Forex
+import com.snowplowanalytics.forex._
+import com.snowplowanalytics.forex.oerclient.{OerClientConfig, DeveloperAccount, AccountType, OerResponseError}
+import com.snowplowanalytics.forex.oerclient.OerResponseError._
+import com.snowplowanalytics.forex.{Forex, ForexConfig}
+import com.snowplowanalytics.forex.ForexLookupWhen._
+
+// This project
+import common.utils.ConversionUtils
+import utils.ScalazJson4sUtils
+
+import com.snowplowanalytics.forex.oerclient._
+/**
+* Companion object. Lets us create an CurrencyConversionEnrichment
+* instance from a JValue.
+*/
+object CurrencyConversionEnrichmentConfig extends ParseableEnrichment {
+
+  val supportedSchemaKey = SchemaKey("com.snowplowanalytics.snowplow", "currency_conversion_config", "jsonschema", "1-0-0")
+  
+  //Creates an CurrencyConversionEnrichment instance from a JValue
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[CurrencyConversionEnrichment] = {
+    isParseable(config, schemaKey).flatMap( conf => {
+      (for {
+        apiKey        <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+// Object and a case object with the same name
+case class CurrencyConversionEnrichment(
+  apiKey: String,
+  bsCurrency: String, 
```

Comment:
"bs" isn't a good abbrev. in English (http://www.urbandictionary.com/define.php?term=BS). Use `baseCurrency` here, that's fine.